x/nexus/keeper/lockable_asset.go

```diff
@@ -176,11 +172,13 @@ func mint(ctx sdk.Context, bank types.BankKeeper, toAddr sdk.AccAddress, coin sd
 	return nil
 }
 
-func getCoinType(ctx sdk.Context, nexus types.Nexus, denom string) (types.CoinType, error) {
+func getCoinType(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, denom string) (types.CoinType, error) {
 	switch {
 	// check if the format of token denomination is 'ibc/{hash}'
 	case isIBCDenom(denom):
 		return types.ICS20, nil
+	case isFromExternalCosmosChain(ctx, nexus, ibc, denom):
```

Comment:
```suggestion
    // check if the denom is the registered asset name for an ICS20 coin from a cosmos chain
	case isFromExternalCosmosChain(ctx, nexus, ibc, denom):
```
would be nice to make this as clear as possible